pkg/platforms/openshift/openshift.go

```diff
@@ -22,29 +32,36 @@ const (
 //go:generate ../../../bin/mockgen -destination mock/mock_openshift.go -source openshift.go
 type OpenshiftContextInterface interface {
 	GetFlavor() OpenshiftFlavor
-	GetMcClient() mcclientset.Interface
 	IsOpenshiftCluster() bool
 	IsHypershift() bool
+
+	OpenshiftDrainNode(context.Context, *corev1.Node) (bool, error)
```

Comment:
nit: names for the drain* functions are a bit confusing. My original thought was that these functions will do complete drain operation for Openshift cluster. Maybe we can rename functions to `OpenshiftBeforeDrainNode` and OpenshiftAfterDrainNode` to make it clear that these functions provide  extra logic for the drain, but doesn't do drain by their own. WDYT?

Reply:
no problem sounds good to me